mmv1/third_party/terraform/services/container/node_config.go.erb

```diff
@@ -101,21 +101,22 @@ func schemaLoggingVariant() *schema.Schema {
 }
 
 func schemaGcfsConfig() *schema.Schema {
-        return &schema.Schema{
-                Type:     schema.TypeList,
-		Optional: true,
-		MaxItems: 1,
+	return &schema.Schema{
+		Type:        schema.TypeList,
```

Comment:
I can give it a shot and see how the tests (and functional testing) look; I'm Ok with it as long as everything still works.

That said, from what I understand, empty blocks are legal Terraform (and we've seen cases, like in `terraform-google-modules/terraform-google-kubernetes-engine` where there's a lot of templating, dynamic blocks, etc that this sometimes is a thing)? See for example the sample code in https://github.com/hashicorp/terraform-provider-google/issues/19428 with an empty `node_pool_defaults.node_config_defaults` block... I think the requirement to have at least one field in a nested block used to be a common practice, but that there are now better safeguards? I doubt other fields would be added to this block, but #11572 is an example of a field that was defined as required more or less arbitrarily at one point in the past, and ended up causing issues.

FWIW, in the functional testing I did, I'm pretty sure I tested the behavior with that block empty and other scenarios, and didn't see any issues with it, though of course it's possible I missed something.

You would probably know better, but what effect does making a top level field `Computed` do exactly, since it doesn't have any value of its own? Does it just get inherited by everything underneath?

Reply:
Basically like this, plus a docs update, is what you're thinking?

```diff
diff --git a/mmv1/third_party/terraform/services/container/node_config.go.erb b/mmv1/third_party/terraform/services/container/node_config.go.erb
index 94e1e04c0..df45e8d50 100644
--- a/mmv1/third_party/terraform/services/container/node_config.go.erb
+++ b/mmv1/third_party/terraform/services/container/node_config.go.erb
@@ -104,14 +104,14 @@ func schemaGcfsConfig() *schema.Schema {
 	return &schema.Schema{
 		Type:        schema.TypeList,
 		Optional:    true,
+		Computed:    true,
 		MaxItems:    1,
 		Description: `GCFS configuration for this node.`,
 		Elem:        &schema.Resource{
 			Schema: map[string]*schema.Schema{
 				"enabled": {
 					Type:        schema.TypeBool,
-					Optional:    true,
-					Computed:    true,
+					Required:    true,
 					Description: `Whether or not GCFS is enabled`,
 				},
 			},
```

Reply:
I can validate that at a basic level, it has the effect of preventing `enabled` from being skipped.  I can also create the cluster without it set, add it later, and even remove the block after applying (in which case it's a noop -- terraform will get the existing computed value but won't try to set it)

I'll complete running the tests locally, and push up this change as a new commit if it passes, though would appreciate if you're able to expedite any additional tests / review on your end if we go this way (esp. since I've got another change in the same area of the codebase that may end up having some conflicts when one or the other goes in)

```
│ Error: Missing required argument
│ 
│   on cluster.tf line 7, in resource "google_container_cluster" "test_gcfs_config":
│    7:     gcfs_config {
│ 
│ The argument "enabled" is required, but no definition was found.
```

Tests pass, at least for `TestAccContainerCluster_withNodeConfigGcfsConfig`:
```
--- PASS: TestAccContainerCluster_withNodeConfigGcfsConfig (931.54s)
PASS
ok  	github.com/hashicorp/terraform-provider-google/google/services/container	932.504s
```